### PR TITLE
Use refactorWithScope() in MakeInheritedMethodVisibilitySameAsParentRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php [hunk 1]
```diff
@@ -79,51 +78,56 @@ protected function run()
...
-                return null;
-            }
+            /** @var string $methodName */
+            $methodName = $this->getName($classMethod->name);
 
-            if ($this->isConstructorWithStaticFactory($node, $methodName)) {
-                return null;
-            }
+            foreach ($classReflection->getParents() as $parentClassReflection) {
+                $nativeClassReflection = $parentClassReflection->getNativeReflection();
```
Comment (samsonasik) [inline rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php:104]: The `$classReflection->getParents()` need to be saved to variable before 1st parent loop to increase performance

**Comment by samsonasik** [comment]
the :
```php
if (! $this->phpVersionProvider->isAtLeastPhpVersion(PhpVersionFeature::PARENT_VISIBILITY_OVERRIDE)) {
```

need to be moved up before loop

File: rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php [hunk 2]
```diff
@@ -163,13 +167,7 @@ private function isConstructorWithStaticFactory(ClassMethod $classMethod, string
...
         }
 
-        $classLike = $this->betterNodeFinder->findParentType($classMethod, Class_::class);
-        $classLike = $this->betterNodeFinder->findParentType($classMethod, Class_::class);
-        if (! $classLike instanceof Class_) {
-            return false;
-        }
-
-        foreach ($classLike->getMethods() as $iteratedClassMethod) {
+        foreach ($class->getMethods() as $iteratedClassMethod) {
```
Comment (samsonasik) [inline rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php]: the class->getMethods() need to be moved up to parent loop before 1st loop so no need to call again in 2nd loop

Comment (TomasVotruba) [reply rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php]: I've update the PR :+1:

File: rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php [hunk 3]
```diff
@@ -79,51 +71,54 @@ protected function run()
...
 
-        /** @var string $methodName */
-        $methodName = $this->getName($node->name);
+        if ($classReflection->getParents() === []) {
+            return null;
+        }
 
-        foreach ($classReflection->getParents() as $parentClassReflection) {
-            $nativeClassReflection = $parentClassReflection->getNativeReflection();
+        $parentClassReflections = $classReflection->getParents();
```
Comment (samsonasik) [inline rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php:93]: `$parentClassReflections` can be moved up so no need double call

Comment (samsonasik) [reply rules/CodingStyle/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector.php:93]: https://github.com/rectorphp/rector-src/pull/2246

**Pull request merged**